src/datachain/lib/arrow.py

```diff
@@ -190,18 +225,6 @@ def arrow_type_mapper(col_type: pa.DataType, column: str = "") -> type:  # noqa:
     raise TypeError(f"{col_type!r} datatypes not supported, column: {column}")
 
 
-def _nrows_file(file: File, nrows: int) -> str:
```

Comment:
I have removed this hack. If `nrows` is a significant number, this is likely to be slower. 

We now iterate up to `self.nrows` with the pyarrow dataset.
I tried running `json-csv-reader` example, and the new method is slightly faster.